conclave-enclave/src/main/kotlin/com/r3/conclave/enclave/internal/GramineEntryPoint.kt

```diff
@@ -20,10 +23,12 @@ object GramineEntryPoint {
     private const val EXIT_ERR = -1
 
     /** Enclave metadata, retrieved from the manifest. */
-    private val isSimulation = parseBoolean(System.getenv("CONCLAVE_IS_SIMULATION_ENCLAVE")!!)
+    private val enclaveMode = EnclaveMode.valueOf(System.getenv("CONCLAVE_ENCLAVE_MODE")!!.uppercase())
     private val isPythonEnclave = parseBoolean(System.getenv("CONCLAVE_IS_PYTHON_ENCLAVE")!!)
-    private val signingKeyMeasurement = System.getenv("CONCLAVE_SIMULATION_MRSIGNER")?.let { parseHex(it) }
+    private val isSgxDebug = parseBoolean(System.getenv("SGX_DEBUG"))
```

Comment:
Isn't this redundant in the presence of `enclaveMode`?

Reply:
It is a parameter we have previously built in the manifest.
This is to make sure that we have an SGX manifest with the SGX debug flag (which is a Gramine parameter) corresponding to the enclave mode that we have in it (which is more a Conclave thing).
It might be redundant, but I like the validation bit.
I imagine a situation where we might build the sgx manifest manually and make mistake when doing it.

Reply:
This is _more_ likely to introduce errors, since now we have two tightly coupled parameters.